stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/RabbitMQTransactionContext.java

```diff
@@ -78,8 +84,26 @@ public XAResource getXAResource() {
 
     /**
      * Handles the transaction block if the context is in transaction.
+     *
+     * @param strand Strand.
      */
-    public void handleTransactionBlock() {
-     // TODO: Transaction handling logic
+    public void handleTransactionBlock(Strand strand) {
+        boolean isInTransaction = strand.isInTransaction();
```

Comment:
We are already checking this in outer method before calling this method. So why need this check again?